### ISSUE TITLE
Add uploaded file component 

### ISSUE THREAD

**Issue by racheledelman** [codeforamerica/honeycrisp-gem#242]
Add a component for uploaded files (or update one if there's an existing one, but I didn't see it). GCF, GYR, and probably SHIBA have similar components, this is a style update. 

# Behavior
- Doesn't appear on the page until the first file has been selected
- During upload, a blank thumbnail appears with a spinner (steal from GYR), and the words "Attaching photo..."
- After upload, file appears with name and thumbnail
- "Remove" deletes the file
- Copy error behavior from one of the existing projects  

# Spec
![image](https://user-images.githubusercontent.com/2693690/109094991-c4620c80-76cf-11eb-8521-254c4b632e26.png)

# Mockup
![image](https://user-images.githubusercontent.com/2693690/109095415-87e2e080-76d0-11eb-894b-90f4c4de721e.png)

**Comment by bengolder** [comment]
💯 Love that this is being addressed. In some cases we are not able to generate a thumbnail for the uploaded file. What should we show if we can't make a thumbnail?

**Comment by racheledelman** [comment]
Good point, Ben. Let's show an [image icon](http://honeycrisp.herokuapp.com/cfa/styleguide#atoms-icons).

![image](https://user-images.githubusercontent.com/2693690/109098865-f4f97480-76d6-11eb-9206-18a172457917.png)
